docs/features/enumerations.rst

```diff
@@ -0,0 +1,73 @@
+Enumerations
+============
+
+Description
+-----------
+
+DSG can handle the generation of enumerations in your `.proto` files in two ways:
+
+- Using an Annotated Serializer or Model field, with choices
+- Using FieldDict in the `grpc_action` decorator
+
+Unless you are using enums in a FieldDict directly, you should be using `TextChoices` or `IntegerChoices` as defined in the Django documentation `here <https://docs.djangoproject.com/en/5.1/ref/models/fields/#enumeration-types>`_.
+
+Using a FieldDict
+-----------------
+
+In a FieldDict, you can specify an Enum for the type. This will generate the corresponding enum in the `.proto` file.
+
+.. code-block:: python
+
+    @grpc_action(
+        request=[{"name": "enum_example", "type": MyEnum}],
+        response=[{"name": "value", "type": "string"}],
+    )
+
+Using an Annotated Serializer or Model field
+--------------------------------------------
+
+If you want to use choices in a Serializer or Model, you can use `TextChoices` or `IntegerChoices` and use `Annotated` to specify the choices you use as metadata.
+
+DSG is not able to directly access the choices in the field as they are normalized as a tuple missing the keys; that's why we use `Annotated`.
+
+.. warning::
+
+    If you don't use `Annotated`, no enum will be generated in the `.proto` file; instead, your field will simply be a string or int32 type.
+
+Example in a model:
+
+.. code-block:: python
```

Comment:
add the imports, annotated is not that known
Or refer to it as `typing.Annotated` which is clearer